packages/preview-middleware/src/base/flp.ts

```diff
@@ -0,0 +1,231 @@
+            if (app.local) {
+                const manifest = JSON.parse(readFileSync(join(app.local, 'webapp/manifest.json'), 'utf-8'));
+                this.addApp(manifest, app);
+                this.router.use(app.target, serveStatic(join(app.local, 'webapp')));
+                this.logger.info(`Serving additional application at ${app.target} from ${app.local}`);
+            }
+        }
+    }
+
+    /**
+     * Create required routes for flex.
+     *
+     * @returns template configuration for flex.
+     */
+    private createFlexHandler(): TemplateConfig['ui5']['flex'] {
+        const workspaceConnectorPath = '/preview/WorkspaceConnector';
+        this.router.get(`/resources${workspaceConnectorPath}.js`, (_req: Request, res: Response) => {
+            res.status(200)
+                .contentType('text/javascript')
+                .send(readFileSync(join(__dirname, '../../templates/flp/workspaceConnector.js'), 'utf-8'));
+        });
+        const api = '/preview/api/changes';
+        this.router.use(api, json());
+        this.router.get(api, async (_req: Request, res: Response) => {
+            res.status(200)
+                .contentType('application/json')
+                .send(await readChanges(this.project, this.logger));
+        });
+        this.router.post(api, async (req: Request, res: Response) => {
+            try {
+                const { success, message } = writeChange(
+                    req.body,
+                    this.utils.getProject().getSourcePath(),
+                    this.logger
+                );
+                if (success) {
+                    res.status(200).send(message);
+                } else {
+                    res.status(400).send('INVALID_DATA');
+                }
+            } catch (error) {
+                res.status(500).send(error.message);
+            }
+        });
+        this.router.delete(api, async (req: Request, res: Response) => {
+            try {
+                const { success, message } = deleteChange(
+                    req.body,
+                    this.utils.getProject().getSourcePath(),
+                    this.logger
+                );
+                if (success) {
+                    res.status(200).send(message);
+                } else {
+                    res.status(400).send('INVALID_DATA');
+                }
+            } catch (error) {
+                res.status(500).send(error.message);
+            }
+        });
+
+        return [
+            {
+                applyConnector: workspaceConnectorPath,
+                writeConnector: workspaceConnectorPath,
+                custom: true
+            }
+        ];
+    }
+
+    /**
+     * Add an application to the local FLP preview.
+     *
+     * @param manifest manifest of the additional target app
+     * @param app configuration for the preview
+     */
+    addApp(manifest: Manifest, app: App) {
+        const id = manifest['sap.app'].id as string;
```

Comment:
we can remove `as string`. We don't need the assertion.

Reply:
https://github.com/SAP/open-ux-tools/pull/1093/commits/0c0b26cacf59f4ead932600574c40e3cfafbe2f2

Reply:
thx